python/cuml/preprocessing/TargetEncoder.py

```diff
@@ -0,0 +1,213 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf as gd
+import cupy as cp
+from cuml.common.exceptions import NotFittedError
+
+
+class TargetEncoder:
+    """
+    A cudf based implementation of target encoding
```

Comment:
Should include a definition of target encoding here. Ideally a link to a paper or blog for more details, but the definition is most important.